sigllm/data.py

```diff
@@ -0,0 +1,183 @@
+    """
+    timestamp_column_name = df.columns[timestamp_column] if timestamp_column else df.columns[0]
+    value_column_names = df.columns[value_columns] if value_columns else df.columns[1:]
+
+    data = dict()
+    data['timestamp'] = df[timestamp_column_name].astype('int64').values
+    for column in value_column_names:
+        data[column] = df[column].astype(float).values
+
+    return pd.DataFrame(data)
+
+
+def load_csv(path, timestamp_column=None, value_column=None):
+    """Load and format CSV file.
+
+    Args:
+        path (str): Path to CSV file
+        timestamp_column: Column index or name for timestamp
+        value_column: Column index or name for values
+
+    Returns:
+        pd.DataFrame: Loaded and formatted DataFrame
+
+    Raises:
+        ValueError: If column specifications are invalid
+    """
+    header = None if timestamp_column is not None else 'infer'
+    data = pd.read_csv(path, header=header)
+
+    if timestamp_column is None:
+        if value_column is not None:
+            raise ValueError('If value_column is provided, timestamp_column must be as well')
+        return data
+
+    elif value_column is None:
+        raise ValueError('If timestamp_column is provided, value_column must be as well')
+    elif timestamp_column == value_column:
+        raise ValueError('timestamp_column cannot be the same as value_column')
+
+    return format_csv(data, timestamp_column, value_column)
+
+
+def load_normal(
+    name, timestamp_column=None, value_column=None, start=None, end=None, use_timestamps=False
+):
+    """Load normal data from file or download if needed.
+
+    Args:
+        name (str):
+            Name or path of the normal data.
+        timestamp_column (str or int):
+            Column index or name for timestamp.
+        value_column (str or int):
+            Column index or name for values.
+        start (int or timestamp):
+            Optional. If specified, this will be start of the sub-sequence.
+        end (int or timestamp):
+            Optional. If specified, this will be end of the sub-sequence.
+        use_timestamps (bool):
+            If True, start and end are interpreted as timestamps.
+            If False, start and end are interpreted as row indices.
+
+    Returns:
+        pandas.DataFrame:
+            Loaded subsequence with `timestamp` and `value` columns.
+    """
+    if os.path.isfile(name):
+        data = load_csv(name, timestamp_column, value_column)
+    else:
+        data = download_normal(name)
+
+    data = format_csv(data)
+
+    # Handle slicing if start or end is specified
+    if start is not None or end is not None:
+        if use_timestamps:
+            # If start and end are timestamps
+            mask = (data['timestamp'] >= start) & (data['timestamp'] <= end)
+            data = data[mask]
+        else:
+            # If start and end are indices
+            data = data.iloc[start:end]
```

Comment:
couple of things
* we don't know the exact name of the timestamp column, so you should use the variable.
* the user typically is not keep on specifying so many parameters. we can check if start or end are integers then we can use them to slice. Otherwise, assume that they are timestamps.
* check if the edge cases `start=None` but `end` has a value works and vice versa.  

```python3
if start is not None or end is not None:
    if isinstance(start, int) or isinstance(end, int):
        # If start and end are indices
        data = data.iloc[start:end]
    else:
        # If start and end are timestamps
        mask = (data[timestamp_column] >= start) & (data[timestamp_column] <= end)
        data = data[mask]
```